react/src/views/Transfers/CreateTransferShipment/CreateTransferShipmentView.test.tsx

```diff
@@ -0,0 +1,235 @@
+});
+
+// Test case 4.3.3
+it("4.3.3 - Click on Submit Button", async () => {
+  const user = userEvent.setup();
+  render(<CreateTransferShipmentView />, {
+    routePath: "/bases/:baseId/transfers/shipments/create",
+    initialUrl: "/bases/1/transfers/shipments/create",
+    additionalRoute: "/bases/1/transfers/shipments",
+    mocks: [initialQuery, successfulMutation],
+    addTypename: true,
+    globalPreferences: {
+      dispatch: jest.fn(),
+      globalPreferences: {
+        selectedOrganisationId: organisation1.id,
+        availableBases: organisation1.bases,
+      },
+    },
+  });
+  const title = await screen.findByRole("heading", { name: "Start New Shipment" });
+  expect(title).toBeInTheDocument();
+
+  // Test case 4.3.3.1 - Form data was valid and mutation was successful
+
+  const submitButton = await screen.findByRole("button", { name: /start/i });
+  expect(submitButton).toBeInTheDocument();
+
+  await assertOptionsInSelectField(user, /organisation/i, [/boxcare/i], title);
+  await selectOptionInSelectField(user, /organisation/i, "BoxCare");
+  expect(await screen.findByText("BoxCare")).toBeInTheDocument();
+  await assertOptionsInSelectField(user, /base/i, [/samos/i, /thessaloniki/i, /athens/i], title);
+  await selectOptionInSelectField(user, /base/i, "Samos");
+  expect(await screen.findByText("Samos")).toBeInTheDocument();
+
+  await user.click(submitButton);
+  expect(await screen.findByText(/successfully created a new shipment/i)).toBeInTheDocument();
+  // Test case 4.3.3.2 - Redirect to Transfers Shipments Page
+  expect(
+    await screen.findByRole("heading", { name: "/bases/1/transfers/shipments" }),
+  ).toBeInTheDocument();
+
+  cleanup();
+  render(<CreateTransferShipmentView />, {
+    routePath: "/bases/:baseId/transfers/shipments/create",
+    initialUrl: "/bases/1/transfers/shipments/create",
+    mocks: [initialQuery, mutationNetworkError],
+    addTypename: true,
+    globalPreferences: {
+      dispatch: jest.fn(),
+      globalPreferences: {
+        selectedOrganisationId: organisation1.id,
+        availableBases: organisation1.bases,
+      },
+    },
+  });
+
+  // Test case 4.3.3.3 - Form data was valid, but the mutation failed
+  const pageTitle = await screen.findByRole("heading", { name: "Start New Shipment" });
+  expect(pageTitle).toBeInTheDocument();
+
+  const submitStartButton = await screen.findByRole("button", { name: /start/i });
+  expect(submitStartButton).toBeInTheDocument();
+
+  await assertOptionsInSelectField(user, /organisation/i, [/boxcare/i], pageTitle);
+  await selectOptionInSelectField(user, /organisation/i, "BoxCare");
+  expect(await screen.findByText("BoxCare")).toBeInTheDocument();
+  await selectOptionInSelectField(user, /base/i, "Samos");
+  expect(await screen.findByText("Samos")).toBeInTheDocument();
+  await user.click(submitStartButton);
+  expect(await screen.findByText(/your changes could not be saved!/i)).toBeInTheDocument();
```

Comment:
I added a new test case 4.3.3.4 to the test plan. It's similar to 4.3.3.3 but the `errors` list in the mutation response needs to be filled out (to mock an error in the BE during processing the request). Can you implement this?